### ISSUE TITLE
External redirects broken with Vercel adapter

### ISSUE THREAD

**Issue by wotschofsky** [withastro/astro#11313]
### Astro Info

```block
Astro                    v4.11.0
Node                     v18.20.3
System                   Linux (x64)
Package Manager          npm
Output                   server
Adapter                  @astrojs/vercel/serverless
Integrations             none

(from CodeSandbox)
```

### Describe the Bug

Redirects defined in `astro.config.mjs` are improperly transformed by the Vercel adapter during build, making it impossible to redirect to  an external URL.

Example: `https://google.com` -> `/https:/google.com` 

The adapter [always runs the redirect destination through Node's `path.join`](https://github.com/withastro/astro/blob/ae3b0968925251e2c56b624c51ff72270d4241ca/packages/integrations/vercel/src/lib/redirects.ts#L87) while prepending the projects base path (usually `/`). Therefore external redirects are "destroyed" and converted into local redirects like shown above.

The final redirects for deployment are defined inside of `.vercel/output/config.json`

See https://vercel-redirect-repro.vercel.app for deployed example

### What's the expected result?

Automatically detect external redirects, for example based on whether a protocol is defined, and skip any transformation logic to preserve the original redirect

### Link to Minimal Reproducible Example

https://codesandbox.io/p/devbox/hopeful-scooby-cpzf5m

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by ematipico** [comment]
I don't think this is a bug, our redirects, by design, aren't meant to handle external redirects

**Comment by wotschofsky** [comment]
I see, wasn't aware of this.
Given that the Cloudflare adapter supports external redirects, I was assuming external redirects should be supported.

But if the platform supports external redirects, why not allow them to be configured adapter-agnosticly through the Astro config?

**Comment by ematipico** [comment]
Because Astro doesn't know which platform you will use, and that's where the adapter comes into play. We could consider e feature request for the Vercel adapter, it that's doable

**Comment by wotschofsky** [comment]
I believe this would be a rather simple change, which I'd happily implement
